src/runtime/hashmap.go

```diff
@@ -356,6 +356,24 @@ func hashmapStringDelete(m *hashmap, key string) {
 //go:linkname valueInterfaceUnsafe reflect.valueInterfaceUnsafe
 func valueInterfaceUnsafe(v reflect.Value) interface{}
 
+func hashmapFloat32Hash(ptr unsafe.Pointer) uint32 {
+	f := *(*uint32)(ptr)
+	if f == 0x80000000 {
+		// convert -0 to 0 for hashing
+		f = 0
+	}
+	return hashmapHash(unsafe.Pointer(&f), 4)
+}
```

Comment:
The IR for this looks pretty neat! Translated back to pseudo-Go, it's this:

```go
var stackalloc [1]uint64 // uninitialized stack allocation
stackalloc[0] = 0
val := *ptr
comparison = val == -2147483648
val = comparison ? 0 : val // this is a single LLVM IR instruction
*(*uint32)(&stackalloc) = val
return hashmapHash(&stackalloc, 4)
```

(the `[1]uint64` is because of reasons, it should be optimized)

Reply:
https://github.com/tinygo-org/tinygo/pull/2362